Detections/MultipleDataSources/AAD_PAVPN_Correlation.yaml

```diff
@@ -0,0 +1,56 @@
+id: ba144bf8-75b8-406f-9420-ed74397f9479
+name: IP with multiple failed Azure AD logins successfully logs in to Palo Alto VPN.
+description: |
+    This query creates a list of IP addresses with a number failed login attempts to AAD 
+    above a set threshold.  It then looks for any successful Palo Alto VPN logins from any
+    of these IPs within the same timeframe.
+severity: Medium
+requiredDataConnectors:
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+     - SigninLogs
+  - connectorId: PaloAltoNetworks
+    dataTypes:
+     - CommonSecurityLog
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+  - CredentialAccess
+relevantTechniques:
+  - T1078
+  - T1110
+
+query: |
+  //Set a threshold of failed AAD signins from an IP address within 1 day above which we want to deem those logins suspicious.
+  let signin_threshold = 5; 
```

Comment:
set the timeframe at top - 
let timeframe = 1d; 
and then put timeframe in as the variable where you have 1d for example
| where TimeGenerated >= ago(timeframe)

Reply:
Done